packages/augur-ui/src/modules/create-market/get-template.ts

```diff
@@ -0,0 +1,158 @@
+
+export type UserInputtedType =
+  | UserInputText
+  | UserInputDateYear
+  | UserInputDateTime
+  | UserInputDropdown
+  | UserInputUserOutcome;
+
+interface CategoryTemplate {
+  templates: Template[];
+  children: CategoryTemplate[];
+}
+export interface Template {
+  categories: Categories;
+  marketType: string;
+  question: string;
+  example: string;
+  inputs: TemplateInput[];
+  resolutionRules: string[];
+}
+
+export interface TemplateInput {
+  type: TemplateInputType;
+  placeholder: string;
+  tooltip?: string;
+  userInput: UserInputtedType;
+}
+
+export interface Categories {
+  primary: string;
+  secondary: string;
+  tertiary: string;
+}
+
+export const getTemplates = (
+  categories: Categories,
+  marketType: string
+): Template[] => {
+  if (!marketType) return [];
+  let categoryTemplates = templates[categories.primary];
+
+  if (!categoryTemplates) return [];
+  if (!categories.secondary)
+    return getTemplatesByMarketType(categoryTemplates.templates, marketType);
+
+  categoryTemplates = categoryTemplates[categories.secondary];
+  if (!categoryTemplates) return [];
+  if (!categories.tertiary)
+    return getTemplatesByMarketType(categoryTemplates.templates, marketType);
+
+  categoryTemplates = categoryTemplates[categories.tertiary];
+  return getTemplatesByMarketType(categoryTemplates.templates, marketType);
+};
+
+const getTemplatesByMarketType = (
+  categoryTemplates: Template[],
+  marketType
+) => {
+  return categoryTemplates.filter(c => c.marketType === marketType);
+};
+
+const templates = {
+  sports: {
+    templates: [],
+    children: {
+      soccer: {
+        templates: [
+          {
+            marketType: CATEGORICAL,
+            question: `Which team will win: [0] vs [1], [2]`,
+            example: `Which team will win: Real Madrid vs Manchester United, Estimated schedule start time: Sept 19, 2019 8:20 pm EST`,
+            inputs: [
+              {
+                type: TemplateInputType.USER_OUTCOME,
+                placeholder: `Team A`,
+              },
+              {
+                type: TemplateInputType.USER_OUTCOME,
+                placeholder: `Team B`,
+              },
+              {
+                type: TemplateInputType.DATETIME,
+                placeholder: `Estimated scheduled start time`,
+              },
+              {
+                type: TemplateInputType.ADDED_OUTCOME,
+                placeholder: `Other`,
+                tooltip: `A team not listed as an outcome wins the event`,
+              },
+            ],
+            resolutionRules: [
+              ` If the game is NOT played or is not deemed an official game, meaning, less than 90% of the scheduled match had been completed, or ends in a tie, the market should resolve as "Draw/No Winner".`,
+            ],
+          },
+          {
+            marketType: CATEGORICAL,
+            question: `[0] vs [1]: Total goals scored; Over/Under [2].5`,
```

Comment:
Can you use a key instead of index

Reply:
added an id to the inputs. like a key but called `id`